substrate/frame/contracts/src/exec.rs

```diff
@@ -3113,6 +3163,62 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn read_only_call_with_non_zero_value_fails() {
+		let code_bob = MockLoader::insert(Call, |ctx, _| {
+			ctx.ext.call(
+				Weight::zero(),
+				BalanceOf::<Test>::zero(),
+				CHARLIE,
+				ctx.input_data[0] as u64,
+				vec![],
+				true,
+				true,
+			)
+		});
+
+		let code_charlie = MockLoader::insert(Call, |_, _| exec_success());
+
+		ExtBuilder::default().build().execute_with(|| {
+			let schedule = <Test as Config>::Schedule::get();
+			place_contract(&BOB, code_bob);
+			place_contract(&CHARLIE, code_charlie);
+			let contract_origin = Origin::from_account_id(ALICE);
+			let mut storage_meter =
+				storage::meter::Meter::new(&contract_origin, Some(0), 0).unwrap();
+
+			// BOB calls CHARLIE with read-only flag and zero value
+			assert_ok!(MockStack::run_call(
+				contract_origin.clone(),
+				BOB,
+				&mut GasMeter::<Test>::new(GAS_LIMIT),
+				&mut storage_meter,
+				&schedule,
+				0,
+				vec![0],
+				None,
+				Determinism::Enforced
+			));
+
+			// BOB calls CHARLIE with read-only flag and non zero value
+			assert_err!(
+				MockStack::run_call(
+					contract_origin,
+					BOB,
+					&mut GasMeter::<Test>::new(GAS_LIMIT),
+					&mut storage_meter,
+					&schedule,
+					0,
+					vec![1],
```

Comment:
What's happening here? `value` is `0` but `input_data` is set?

Reply:
BOB uses the input provided here as a value when calling CHARLIE.